### PR TITLE
Metrics updates

### LINKED ISSUE #3065: Update info on buffered flag support for long tasks
**Describe the bug**
At https://web.dev/custom-metrics/#long-tasks-api it says:

"Caution: The buffered flag does not currently work for Long Tasks (though support is being added). In the meantime, you can track Long Tasks by registering the PerformanceObserver in the <head> of your pages, before loading any other scripts."

Chrome has been updated to support the buffered flag for Long Tasks. Can the wording be updated?

### LINKED ISSUE #3108: content: Small suggestions for LCP
Feedback for: https://web.dev/lcp/. Overall the page is great so I only have some small suggestions.

* In the sentence "The Largest Contentful Paint (LCP) metric reports the render time of the largest content element visible within the viewport.", maybe note that LCP is restricted to text and image content?

* We note that removed elements stop counting as LCP candidates. I wonder if it's worth noting that this might change in the future due to image carousel problems?

* In the sample code, after the line `po.takeRecords().forEach(updateLCP);`, should the observer be disconnected?

cc @philipwalton

### LINKED ISSUE #3157: content: clarify that FID only measures event queuing time
### What page(s) need to be updated?

https://web.dev/fid/

### Why is this update needed?

Some developers are confused about the definition of FID—and assume it includes the event processing time—so we should be clear that it only includes the event queuing time.

### LINKED ISSUE #3197: LCP article is a little ambiguous
The article contains this section (emphasis added by me):

> To handle this potential for change, the browser dispatches a PerformanceEntry of type largest-contentful-paint identifying the largest contentful element as soon as the browser has painted the first frame. But then, after rendering subsequent frames, it will dispatch another PerformanceEntry any time the largest contentful element **changes**.

I was a little unclear on what "changes" means in this context. One interpretation is that it means if the largest element changes to another element. Another interpretation is that it means if the largest element itself changes in some way (i.e. repaints, gets resized, etc). Or perhaps it means both?

### LINKED ISSUE #3237: Update LCP defined article for inaccuracy in wording
**Describe the bug**
On this LCP defined page it mentions there are three factors to improve LCP but then lists four:

How to improve LCP #
LCP is primarily affected by three factors:

Slow server response times
Render-blocking JavaScript and CSS
Resource load times
Client-side rendering

**To Reproduce**
1. Visit https://web.dev/lcp/#how-to-improve-lcp

**Expected behavior**
Text should say "LCP is primarily affected by four factors:" or one bullet point removed
**Screenshots**

### PR DESCRIPTION
**Pull request by philipwalton**
This PR fixes or addresses a number of recent issues in the metrics documentation:

- Fixes #3065
- Addresses #3067
- Addresses #3068
- Fixes #3108
- Fixes #3197
- Fixes #3157
- Fixes #3237

cc: @npm1, @anniesullie, @robdodson

### CODE REVIEW

File: src/site/content/en/metrics/custom-metrics/index.md [hunk 1]
```diff
@@ -277,6 +270,69 @@ try {
...
 
+### Event Timing API
+
+The [First Input Delay (FID)](/fid/) metric measures the time from when a user
+first interacts with a page to the time when the browser is actually able to
+begin processing event handlers in response to that interaction. However, in
+some cases it may also be useful to measure the event processing time itself as
+well as the time until the next frame can be rendered.
+
+This is possible with the [Event Timing
```
Comment (npm1) [inline src/site/content/en/metrics/custom-metrics/index.md:281]: Drive-by: should we mention when it's shipping, as it's not on Stable yet? M85.

File: src/site/content/en/metrics/custom-metrics/index.md [hunk 2]
```diff
@@ -277,6 +270,69 @@ try {
...
+- [`startTime`](https://w3c.github.io/performance-timeline/#dom-performanceentry-starttime):
+  the time when the browser receives the event.
+- [`processingStart`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingstart):
+  the time when the browser is able to beginning processing event handlers for
+  the event.
+- [`processingEnd`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingend):
+  time when the browser finishing executing all synchronous code initiated from
+  event handlers for this event.
+- [`duration`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingstart):
+  the time between when the browser receives the event until it's able to paint
```
Comment (npm1) [inline src/site/content/en/metrics/custom-metrics/index.md:294]: I think this is not sufficiently clear. The 'init' time is clear but the 'final' one is the next paint after event handlers. Presumably a browser can paint a next frame after receiving an input but without having handled the input.

File: src/site/content/en/metrics/fid/index.md [hunk 3]
```diff
@@ -196,6 +207,24 @@ How you track, report on, and analyze FID will probably be quite a bit different
...
 this.
 
+### Why only consider the input delay?
+
+As mentioned above, FID only measures the "delay" in event processing. It does
+not measure the event processing time itself nor the time it takes the browser
+to update the UI after running event handlers.
+
+Even though this time is important to the user and does affect the experience,
+it's not included in this metric because it would be easy for developers to
```
Comment (npm1) [inline src/site/content/en/metrics/fid/index.md:217]: I think we should phrase this in terms of what improving the metric incentivizes, not what developers can do to 'cheat'. Just a wording suggestion.

File: src/site/content/en/metrics/tbt/index.md [hunk 4]
```diff
@@ -34,7 +37,7 @@ because the browser cannot interrupt a task that's in progress. So in the event
 that a user _does_ interact with the page in the middle of a long task, the
 browser must wait for the task to finish before it can respond.
```
Comment (npm1) [inline src/site/content/en/metrics/tbt/index.md:39]: Should we also incorporate some reasoning re why between FCP and TTI? This can be in a followup. There has been some confusion around the choice of boundaries.

File: src/site/content/en/metrics/lcp/index.md [hunk 5]
```diff
@@ -155,6 +155,8 @@ subsequent frames, it will dispatch another
 [`PerformanceEntry`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry)
 any time the largest contentful element changes.
 
+For example, on a page with text and a hero image the browser may initially just render the text—at which point the browser would dispatch a `largest-contentful-paint` entry whose `element` property likely reference a `<p>` or `<h1>`. Later, once the hero image finishes loading, a second `largest-contentful-paint` entry c be dispatch and its `element` property would reference the `<img>`.
```
Comment (npm1) [inline src/site/content/en/metrics/lcp/index.md:158]:
```suggestion
For example, on a page with text and a hero image the browser may initially just render the text—at which point the browser would dispatch a `largest-contentful-paint` entry whose `element` property would likely reference a `<p>` or `<h1>`. Later, once the hero image finishes loading, a second `largest-contentful-paint` entry would be dispatched and its `element` property would reference the `<img>`.
```

File: src/site/content/en/metrics/custom-metrics/index.md (hunk 1 above)
Comment (philipwalton) [reply src/site/content/en/metrics/custom-metrics/index.md:281]: This one is tricky because the `first-input` part of the Event Timing API does exist, and it's technically what this section is referring to. (But as you point out it won't work for general event timing use cases until M85.)

**Comment by kaycebasques** [comment]
@philipwalton please re-assign to me when all of @npm1's feedback has been resolved

File: src/site/content/en/metrics/custom-metrics/index.md (hunk 2 above)
Comment (philipwalton) [reply src/site/content/en/metrics/custom-metrics/index.md:294]: Done.

File: src/site/content/en/metrics/tbt/index.md (hunk 4 above)
Comment (philipwalton) [reply src/site/content/en/metrics/tbt/index.md:39]: Sure, but let's do it in a followup PR so this content can be published ASAP. Can you file an issue with the suggested changes/reasoning?

File: src/site/content/en/metrics/fid/index.md (hunk 3 above)
Comment (philipwalton) [reply src/site/content/en/metrics/fid/index.md:217]: Updated.

**Comment by philipwalton** [comment]
@kaycebasques I addressed all feedback, PTAL.

File: src/site/content/en/metrics/tbt/index.md (hunk 4 above)
Comment (npm1) [reply src/site/content/en/metrics/tbt/index.md:39]: Filed https://github.com/GoogleChrome/web.dev/issues/3290

File: src/site/content/en/metrics/custom-metrics/index.md [hunk 6]
```diff
@@ -277,6 +270,71 @@ try {
...
+well as the time until the next frame can be rendered.
+
+This is possible with the [Event Timing
+API](https://wicg.github.io/event-timing/) (which is used to measure FID) as it
+exposes a number of timestamps in the event lifecycle, including:
+
+- [`startTime`](https://w3c.github.io/performance-timeline/#dom-performanceentry-starttime):
+  the time when the browser receives the event.
+- [`processingStart`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingstart):
+  the time when the browser is able to beginning processing event handlers for
```
Comment (kaycebasques) [inline src/site/content/en/metrics/custom-metrics/index.md:288]:
```suggestion
  the time when the browser is able to begin processing event handlers for
```

File: src/site/content/en/metrics/custom-metrics/index.md [hunk 7]
```diff
@@ -277,6 +270,71 @@ try {
...
+API](https://wicg.github.io/event-timing/) (which is used to measure FID) as it
+exposes a number of timestamps in the event lifecycle, including:
+
+- [`startTime`](https://w3c.github.io/performance-timeline/#dom-performanceentry-starttime):
+  the time when the browser receives the event.
+- [`processingStart`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingstart):
+  the time when the browser is able to begin processing event handlers for
+  the event.
+- [`processingEnd`](https://wicg.github.io/event-timing/#dom-performanceeventtiming-processingend):
+  time when the browser finishing executing all synchronous code initiated from
```
Comment (kaycebasques) [inline src/site/content/en/metrics/custom-metrics/index.md:291]:
```suggestion
  time when the browser finishes executing all synchronous code initiated from
```

**Comment by philipwalton** [comment]
Thanks for catching and fixing those typos @kaycebasques!
